File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/model/user/SocialAccount.java
package com.thanh0x.coursedeal.model.user;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Entity representing a linked social account (Google, Apple).
 */
@Entity
@Table(name = "user_social_accounts", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"provider", "providerId"})
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SocialAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private AuthProvider provider;

    private String providerId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @ToString.Exclude
    private UserEntity user;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/model/user/SocialAccount.kt
package com.thanh0x.coursedeal.model.user

import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint

/**
 * Entity representing a linked social account (Google, Apple).
 */
@Entity
@Table(
    name = "user_social_accounts",
    uniqueConstraints = [UniqueConstraint(columnNames = ["provider", "providerId"])],
)
class SocialAccount(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @Enumerated(EnumType.STRING)
    var provider: AuthProvider? = null,
    var providerId: String? = null,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: UserEntity? = null,
)
